docs/MIGRATION_TO_V6.md

```diff
@@ -0,0 +1,205 @@
+
+# Ogmigo v6 migration guide
+
+This document discusses how to upgrade code that uses Ogmigo that is connected to Ogmios v5 instances and must connect to Ogmios v6. While not completely pain-free, it is possible to safely migrate with a minimum of pain.
+
+Note that this document is a supplement to [the original Ogmios migration guide](https://github.com/CardanoSolutions/ogmios/blob/master/architectural-decisions/accepted/017-api-version-6-major-rewrite.md). It’s highly recommended that you read that particular guide before reading this guide, especially if your code handles a lot of individual values. There are also Ogmios API schema pages for [v5](https://ogmios.dev/api/v5.6/) and [v6](https://ogmios.dev/api/) that are worth consulting if all else fails.
+
+# Caveats
+
+There are a handful of caveats that should be considered before reading the main document.
+
+* The Ogmigo v6 upgrade adds no new functionality beyond v6 struct support and a compatibility layer that allows both v5 and v6 Ogmigo JSON output to be unmarshalled into v6 structs. If anything wasn’t supported in Ogmigo v5, it won’t be supported in Ogmigo v6.
+* Support for the Byron era is limited in Ogmigo v6, and may not work as expected. In fact, as of this writing (Nov. 2023), critical functionality, such as the _compatibility_ module, assumes Byron _isn’t_ supported. If you must have Byron support, further Ogmigo work will be required.
+* Any attempt to use a v6-enabled Ogmigo library will automatically break the code using Ogmigo. Many fundamental structs (e.g., _Block_) have been altered to assume v6 structs. If somebody absolutely must use v5 structs, they’ll have to import the _v5_ module and use those structs. Even then, it is highly recommended to use the _compatibility_ module whenever possible, thereby assisting in a smooth transition to the default (v6) code.
+
+# Support for v6
+
+As mentioned in the caveats, the v6 structs are now the default in the Ogmigo library. Code will have to be written with this in mind if working directly with v6. The Ogmios migration guide discusses the gist of what changed between v5 and v6. In general, other than changing specific references within the structs, there aren’t major changes required to understand the new structs.
+
+Note that in particular, Ogmigo automatically handles the switch from JSON-WSP to JSON-RPC 2.0. You should just be able to let Ogmigo handle everything automatically, simply pointing it to the appropriate Ogmios endpoint.
+
+## Major Struct Changes
+
+The Ogmios architectural decision doc linked above isn’t fully correct when discussing how various structs have changed. If you’re using v5 structs and want to see where to look in v6 structs for specific data, you should look at the _compatibility_ module and see how v5 is translated to v6.
```

Comment:
What do you mean here? What isn't "fully correct"? Anything we can fix?

Reply:
Hi! Here's what I found that seemed worth noting, although some of it may have more to do with subtle quirks than being outright wrong. All references are to the v6 structs.

- The v6 API claims `Block.Size` is int64 and not uint64, as in v5. That seems strange but maybe there's a reason for it that I'm missing. There may be other instances but that's the only one I found.
- `Tx.Signatories` can be both `Witness.Bootstrap` and `Witness.Signatories` from v5. There's no explanation regarding this. Is one value present in some eras and not in others?
- Some of the precise JSON formatting has changed. `Tx.TotalCollateral` and `Tx.Fee` are a couple of examples, with `Lovelace` wrapping around the v6 values. Granted, this is an implementation detail and not a fundamental spec issue.